manifest_staging/charts/aad-pod-identity/templates/mic-deployment.yaml

```diff
@@ -23,10 +23,11 @@ spec:
       {{- if .Values.mic.podLabels }}
       {{- toYaml .Values.mic.podLabels | nindent 8 }}
       {{- end }}
-{{- if .Values.mic.podAnnotations }}
       annotations:
-{{ toYaml .Values.mic.podAnnotations | indent 8 }}
-{{- end }}
+        checksum/config: {{ include (print $.Template.BasePath "/mic-secret.yaml") . | sha256sum }}
```

Comment:
This annotation is only required when aad-pod-identity is deployed with an admin secret instead of relying on `azure.json`. So could make this conditional?

`{{- if .Values.adminsecret}}` then set the `checksum/config` annotation?

Reply:
It will make the code a lot less elegant since if there's no pod annotations or admin secret, it'll generate misformatted yaml.  And the checksum will just be a stable value even if adminsecret isn't set so no harm.